File: output/ifc4x3_dev_73740fe4/entity/IfcSurfaceCurve.java
package com.topoplanet.ifc4x3_dev_73740fe4.entity;
import com.topoplanet.test.type.*;


import java.util.List;
import java.util.Set;

public class IfcSurfaceCurve extends IfcCurve {

    public IfcCurve Curve3D;
    public IfcPreferredSurfaceCurveRepresentation MasterRepresentation;
    // INVERSE attributes:
    // BasisSurface : SET [1:2] OF IfcSurface := IfcGetBasisSurface(SELF);
    // INVERSE attributes:
    // CurveIs3D : Curve3D.Dim = 3;
    // CurveIsNotPcurve : NOT ('IFC4X3_DEV_73740fe4.IFCPCURVE' IN TYPEOF(Curve3D));

    // SUPERTYPE OF:
    // - IfcIntersectionCurve
    // - IfcSeamCurve SUBTYPE OF IfcCurve

    // WHERE constraints:
    // CurveIs3D : Curve3D.Dim = 3;
    // CurveIsNotPcurve : NOT ('IFC4X3_DEV_73740fe4.IFCPCURVE' IN TYPEOF(Curve3D));
}

File: src/main/resources/output/IFC4X3_js/Entities/IfcRepresentationItem/IfcGeometricRepresentationItem/IfcCurve/IfcSurfaceCurve/IfcSurfaceCurve.js
class IfcSurfaceCurve extends IfcCurve {
    constructor() {
        /** @type {IFCCURVE} */
        this.Curve3D = null;
        /** @type {IFCPCURVE[]} */
        this.AssociatedGeometry = null;
        /** @type {IFCPREFERREDSURFACECURVEREPRESENTATION} */
        this.MasterRepresentation = null;
    }

    // === EXTENDED BY ===
    // IfcIntersectionCurve
    // IfcSeamCurve

    // === WHERE CLAUSES ===
    // CurveIs3D : Curve3D.Dim = 3
    // CurveIsNotPcurve : NOT ('IFC4X3_DEV_73740fe4.IFCPCURVE' IN TYPEOF(Curve3D))

    // === DERIVE CLAUSES ===
    // BasisSurface : SET [1:2] OF IfcSurface := IfcGetBasisSurface(SELF)
}
